### DISCUSSION TITLE
[5.1] value of an overridden field

### DISCUSSION THREAD

**Discussion by WHITE-developer** [solspace/craft-freeform#1251] [Questions & Answers] [answered]
In Freeform 3 we renderd the checkboxes with files form a matrixblock. We tried to use the way of Freeform 3 where we have an hidden field that gets overridden with a "matrixblock id" and then get the files form that matrixblock. But in Freeform 5 we can't get the value of the overridden field. In the field render the field is filled but when i try to get the "field.value" outside the render the "field.value" is empty.

When we render the field 
`{% if field.handle == "extraData" %} 
    {{ field.render() }} 
{% endif %}`
![image](https://github.com/solspace/craft-freeform/assets/22771556/f258ee4d-ff67-4e10-902c-f8bade7aea22)
The value is filled but when we dump the value of "extraData" it is empty
`{% dd(form.get("extraData").value )%}`
![image](https://github.com/solspace/craft-freeform/assets/22771556/9119651e-9bc8-42c4-a384-ba2dae2e17fb)

And the override looks good 
` fields: { "extraData": {value: blockId,},`

**Steps to reproduce**

1.Upgrade from freeform 3 to 5

**Expected behavior**
Get the value of the overridden field "232"

**Craft & Plugin Info:**

- Craft Version:4.8.10
- Freeform Version:5.1.14
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment, marked as answer]
Hi @WHITE-developer,

Have you seen the updated guide for passing data dynamically to Freeform fields?
https://docs.solspace.com/craft/freeform/v5/guides/passing-dynamic-data-to-forms/

There is a variety of approaches to go about this... one may be better for your specific needs.

If you still have questions, please let me know. 🙂
